### PR TITLE
py-libsonata: new package starting at 0.1.25

### PR DESCRIPTION
**Pull request by tristan0x**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-libsonata/package.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    homepage = "https://github.com/BlueBrain/libsonata"
+    git = "https://github.com/BlueBrain/libsonata.git"
+    pypi = "libsonata/libsonata-0.1.14.tar.gz"
+
+    submodules = True
+
+    maintainers("tristan0x")
+
+    version("develop", branch="master", get_full_repo=True)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-libsonata/package.py]: Is there a reason the name of the version doesn't match that of the branch?  

Normally they are the same.  For more on version comparisons, see https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison

**Review by tldahlgren — COMMENTED** [review]
Confirmed the homepage and versions (sha256 and branch exists).  Do have a concern about the branch version name mismatch.

'spack spec' also succeeds.

File: var/spack/repos/builtin/packages/py-libsonata/package.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    version("0.1.25", sha256="b332efa718123ee265263e1583a5998eaa945a13b8a22903873764cf1d8173fa")
+
+    depends_on("cmake@3.3:", type="build")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-libsonata/package.py]: `CMakeLists.txt` at the source root says the minimum required version is 3.16.  

(Note: That seems to help me get a bit further but I'm still not able to successfully install the package.)

**Review by tldahlgren — COMMENTED** [review]
See my other comment as well.

File: var/spack/repos/builtin/packages/py-libsonata/package.py (hunk 1 above)
Comment (tristan0x) [reply var/spack/repos/builtin/packages/py-libsonata/package.py]: Hello @tldahlgren It thought all development spack versions had to be named `develop`. I updated it to "master"  in 2de655c6743db645ab5621ab705e4f1b49651168

File: var/spack/repos/builtin/packages/py-libsonata/package.py (hunk 2 above)
Comment (tristan0x) [reply var/spack/repos/builtin/packages/py-libsonata/package.py]: I updated the CMake minimum version in b6449ce9480b4b4a263a6f16f3ddc2173333a907
The patch also removes the uses of git submodules in favor of spack packages to fulfill the dependencies.

**Comment by tristan0x** [comment]
Thank you very much for your time @tldahlgren 
I am sorry you were not able to build the package. Could you please share with me the issues you faced?

File: var/spack/repos/builtin/packages/py-libsonata/package.py (hunk 2 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-libsonata/package.py]: In principle, the names should match so someone familiar with the package and or goes to the repository intuitively know which branch they're using.

File: var/spack/repos/builtin/packages/py-libsonata/package.py (hunk 1 above)
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-libsonata/package.py]: In principle, the names should match so someone familiar with the package and or goes to the repository intuitively know which branch they're using.

**Comment by tldahlgren** [comment]
> Thank you very much for your time @tldahlgren I am sorry you were not able to build the package. Could you please share with me the issues you faced?

You're welcome.

The installation of py-cython@3.0.8 is failing for me for `Cython/Compiler/Parsing.c:31:10: fatal error: Python.h: No such file or directory` so it isn't directly in this package.  I've tried other py-cython versions and get different issues so am assuming it's something in my environment.

**Review by tldahlgren — APPROVED** [review]
'spack spec' succeeds.  I am still having problems building `py-cython` but am assuming it's specific to my environment.

**Pull request merged**